### PR TITLE
recreate from create-react-app

### PR DESCRIPTION
**Pull request by ngyukman**

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export interface ISource {
+  repo: string,
+  branch?: string,
+  file?: string,
+  lineNo?: number
+}
+
+const config:any = packageJson;
+export function prefix() { return config.homepage }
+export function gitRestApi() { return config.resetApi }
```
Comment (gluck) [inline src/settings.ts]: resetApi -> restApi

Comment (ngyukman) [reply src/settings.ts]: indeed, fixed

File: src/components/GitCommon.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export function renderNodesForLayout(data: any, layout: string | undefined) {
+  let idx = 0;
+  switch (layout) {
+    case 'google':
+      let headerIdx = 0;
+      const grepNodes = data.map((repo: any) => {
+        return [ (
+            <h4 key={'H'+headerIdx++} className="results">{repo.replace(/\.git$/,'')}</h4>
+          )].concat(data[repo].map((grep: IGrepResultLine) => (
+            <GrepResult key={idx++} repo={grep.repo} branch={grep.branch} file={grep.file} lineNo={grep.lineNo} line={grep.line} layout={layout}/>
```
Comment (gluck) [inline src/components/GitCommon.tsx]: AFAIK REST API returns `line_no`, not `lineNo`

File: src/App.css [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  font-size: 11px;
+  margin: 0;
+}
+
+h4.results {
+    color: white;
+}
+
+pre.results code {
+  white-space: nowrap;
```
Comment (gluck) [inline src/App.css]: AFAIK should be `pre` to retain line start whitespace (gives proper alignment of same file content), but maybe you changed it on purpose ?

**Comment by gluck** [comment]
Other issues found (cf slack):
- background-color required to fix horiz scrolling
- `text-decoration: none` to make the results more readable

Nice work !

**Comment by ngyukman** [comment]
the virtualized list addresses #7 too

**Comment by gluck** [comment]
Looks superb !
```
a:focus, a:hover {
    text-decoration: underline;
}
```
would make the links a bit more natural (and like before) IMO.

**Comment by gluck** [comment]
🎉

**Pull request merged**